edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -84,6 +85,7 @@ public DependencyManager(IConfigurationRoot configuration, X509Certificate2 serv
 
             this.gatewayHostname = Option.Maybe(this.configuration.GetValue<string>(Constants.ConfigKey.GatewayHostname));
             string edgeHubConnectionString = this.configuration.GetValue<string>(Constants.ConfigKey.IotHubConnectionString);
+            this.edgeModuleGenerationId = this.configuration.GetValue<string>(Constants.ConfigKey.ModuleGenerationId);
```

Comment:
nit: can be moved to else because when edgeHubConnectionString is set there won't be a generation id.